### ISSUE TITLE
Add support for loading from pystac objects

### ISSUE THREAD

**Issue by hrodmn** [stac-utils/pgstac#282]
Would it make sense to add support for loading `Collection`, `Item`, and `ItemCollection` objects from `pystac`? It could be useful in python workflows that generate STAC metadata and load directly into `pgstac`.

Right now, I do something like this:
```python
import io
import json

from pypgstac.db import PgstacDB
from pypgstac.load import Loader, Methods
from pystac import ItemCollection
from stactools.package import create_item

# fire up pgstac loader
db = PgstacDB()
loader = Loader(db=db)

# create item collection
item_collection = ItemCollection(
  create_item(href) for href in hrefs
)

# write to ndjson and load into pgstac
buffer = io.BytesIO()
for item in item_collection:
    item.collection_id = COLLECTION_ID_FORMAT.format(
        region=region.value, product=product.value
    )

    buffer.write((json.dumps(item.to_dict()) + "\n").encode("utf-8"))

buffer.seek(0)

loader.load_items(buffer, insert_mode=Methods.upsert)
```

It isn't that hard to write to ndjson and pass that to `load_items`, but it would be nice if that operation was handled by `pypgstac`!

**Comment by keewis** [comment]
looks like you can pass a [sequence of `dict`](https://github.com/stac-utils/pgstac/blob/c7d602a833d29e453309e04e1f47f947d43a34cb/src/pypgstac/python/pypgstac/load.py#L139-L144) to `load_items` and `load_collections`. So this could work already:
```python
items = [item.to_dict() for item in item_collection]
loader.load_items(items, insert_mode=Methods.upsert)
```

**Comment by hrodmn** [comment]
Yeah, that does work. I was thrown off by a bad type hint, I think. Passing a `list` in does in fact work but my type checker complains about `list[Unknown]` being incompatible with `Iterator[Any]`. Maybe this should include `Iterable[Any]` instead of `Iterator[Any]`:
https://github.com/stac-utils/pgstac/blob/c7d602a833d29e453309e04e1f47f947d43a34cb/src/pypgstac/python/pypgstac/load.py#L113

**Comment by keewis** [comment]
true, for `read_json` that could be `Iterable`:  the code even references `Iterable` but does not call `iter()` on the variable. 

However, `Iterator` [inherits](https://docs.python.org/3/library/collections.abc.html#collections-abstract-base-classes) from `Iterable`, and `load_items` at least needs the parameter to be an `Iterator`, so unless you call `read_json` manually you won't be able to pass the narrower type.
